projects/go-lib/src/lib/components/go-file-upload/go-dragon-drop.directive.ts

```diff
@@ -0,0 +1,35 @@
+import { Directive, EventEmitter, HostBinding, HostListener, Output } from '@angular/core';
+
+@Directive({
+  selector: '[goDragonDrop]'
+})
+export class DragonDropDirective {
+  @Output() fileDropped: EventEmitter<any> = new EventEmitter<any>();
+
+  @HostBinding('class.go-file-upload--active') active: boolean = false;
```

Comment:
@StevenUlmer what if we extracted this out into it's own thing? We could pass in the `go-file-upload--active` class as an `@Input` here and then handle the `drop` event in the `file-upload` component as an event handler. 

We don't _have_ to in this PR, could always extract it out later. But that would make this drag/drop functionality reusable which would be nice. 

Thoughts?

Reply:
I went ahead and pulled out al the functionality I could. I had to do some stuff around using class here because just using `@HostBinding('class')` overwrites all classes already applied to the element